Fix  #9: Add Mailcatcher Gem

vanamraghu opened the pull request:
1. Added two gems namely a. Mail catcher b. Thin (added separately as bundle install with only mail catcher is showing compiler errors)
2. Gem file lock file is updated with the newer versions of the dependencies

<!-- Read comments, before committing pull request read checklist again

# Checklist:

- I have performed a self-review of my own code,
- I have commented my code, particularly in hard-to-understand areas,
- Title include "WIP" if work is in progress.

-->

Resolves #9 

### Description
Mailcatcher gem and also thin which is a dependency for mail catcher

### Type of change

<!-- Please delete options that are not relevant. -->

- [ ] Feature

### How to test this PR?

Bundle install should install mail catcher gem and also other dependencies

rt4914 commented on Gemfile:
```diff
@@ -49,6 +49,10 @@ group :development, :test do
+
+  gem 'mailcatcher', '~> 0.2.4'
+
+  gem 'thin', '~> 1.8', '>= 1.8.2'
```
On top of this add a comment. something like this:
`This gem is required with mailcatcher as without this mailcatcher shows <XYZ-error>. The solution was suggestion on https://www.example.com `

This is just a format, fill out details correctly for this.

Following this process gives clear indication why we need this and also helps developers in future to be aware of side-effects because of this gem.

vanamraghu replied:
Provided comments about the error and why separate installation is required

rt4914 replied:
Looks good. Thanks.

rt4914 reviewed (changes requested):
@vanamraghu Suggest change. PTAL (Please take a look)

rt4914 reviewed (changes requested):
@vanamraghu Based on the video that I mentioned in issue and also this page: https://mailcatcher.me/

We should configure this for rails by mentioning the `action_mailer ` properties.

Also update the PR title to `Fix #9: Add Mailcatcher gem`

vanamraghu commented:
> @vanamraghu Based on the video that I mentioned in issue and also this page: https://mailcatcher.me/
> 
> We should configure this for rails by mentioning the `action_mailer ` properties.
> 
> Also update the PR title to `Fix #9: Add Mailcatcher gem`

Updated the title and also added configuration for the mail catcher in development config file

rt4914 reviewed (approved):
LGTM. Thanks.
